### PR TITLE
Notebook orbit5k

### PR DESCRIPTION
**Pull request by nberkouk**
**Reference issues/PRs**

**Types of changes**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Description**

**Screenshots (if appropriate)**

**Any other comments?**

**Checklist**

- [ ] I have read the [guidelines for contributing](https://giotto-ai.github.io/gtda-docs/latest/contributing/#guidelines).
- [ ] My code follows the code style of this project. I used `flake8` to check my Python changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed. I used `pytest` to check this on Python tests.

### CODE REVIEW

File: examples/orbit_5k_pipeline.ipynb [hunk 1]
```diff
@@ -4,18 +4,29 @@
    "cell_type": "markdown",
```
Comment (raphaelreinauer) [inline examples/orbit_5k_pipeline.ipynb:4]: <p>1)Please keep:</p><h4>Author: Raphael Reinauer</h4><p>to make it consistent with the other notebooks</p><p> </p><p>2)Make it clearer:</p><p>consists of subsets of [0, 1]^2, each containing 1000 points.</p><p> </p><p>3) The recursion formula is not correct.</p><p> </p><p>4) We generate S^ρ 1000 times per values of ρ. -&gt; We generate 1000 orbits for each value of rho.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/giotto-ai/giotto-deep/pull/99/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (raphaelreinauer) [inline examples/orbit_5k_pipeline.ipynb:4]: <p>Sort imports</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/giotto-ai/giotto-deep/pull/99/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (raphaelreinauer) [inline examples/orbit_5k_pipeline.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #7.    rho_values = [2.5, 3.5, 4.0, 4.1,4.3]
-&gt; [2.5, 3.5, 4.0, 4.1, 4.3]
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/giotto-ai/giotto-deep/pull/99/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (raphaelreinauer) [inline examples/orbit_5k_pipeline.ipynb:4]: <p>1)\mathbb R is not properly rendered</p><p>2) add reference to persformer paper</p><p>3) &nbsp;<em>It maps a persistence diagram to the logarithm of the class probability.</em> That's not quite true, it's mapping it to the logits.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/giotto-ai/giotto-deep/pull/99/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

**Review by raphaelreinauer — CHANGES_REQUESTED** [review]
Only some small changes.

File: examples/orbit_5k_pipeline.ipynb [hunk 2]
```diff
@@ -4,18 +4,32 @@
...
     "\n",
-    "Depending on the value of the parameter $\\alpha$, the orbits hahve a different dynamics.\n",
+    "1. Start with a uniformly randomly sampled point $(x_0,y_0)\\in [0,1]^2$\n",
+    "2. For $n \\geq 1$, generate $(x_{n+1},y_{n+1})$ inductively by : \n",
     "\n",
-    "The goal of ths notebook is to classify the orbits and infer the parameter $\\alpha$ that was used to generate such orbits."
+    "$$ x_{n+1} = x_{n} + \\rho y_n(1-y_n) ~~~\\text{mod }1$$\n",
+    "$$ y_{n+1} = y_{n} + \\rho x_n(1-x_{n+1}) ~~~\\text{mod }1$$\n",
+    "\n",
+    "3. Define $S^\\rho := \\{(x_i,y_i) \\mid i = 0...999 \\}$\n",
```
Comment (raphaelreinauer) [inline examples/orbit_5k_pipeline.ipynb:20]: In this version it's a 1.
https://github.com/giotto-ai/giotto-deep/blob/notebook_orbit5k/examples/orbit_5k_pipeline.ipynb
image

File: examples/orbit_5k_pipeline.ipynb [hunk 3]
```diff
@@ -121,6 +184,15 @@
...
+    "model_config = PersformerConfig(\n",
+    "    input_size=2 + 2, # there are 2 coordinates and 2 homology dimensions\n",
+    "    ouptut_size=5,  # there are 5 classes\n",
+    "    hidden_size=16,\n",
+    "    intermediate_size=16,\n",
+    "    num_attention_layers=2,\n",
+    "    num_attention_heads=8,\n",
+    ")\n",
+    "\n",
     "wrapped_model = PersformerWrapper(\n",
```
Comment (raphaelreinauer) [inline examples/orbit_5k_pipeline.ipynb:196]: here you redefine the model

File: examples/orbit_5k_pipeline.ipynb [hunk 4]
```diff
@@ -121,6 +184,15 @@
    "source": [
     "# Define the model by using a Wrapper for the Persformer model\n",
     "\n",
+    "model_config = PersformerConfig(\n",
```
Comment (raphaelreinauer) [inline examples/orbit_5k_pipeline.ipynb:187]: Here you define the model config

File: examples/orbit_5k_pipeline.ipynb (hunk 3 above)
Comment (raphaelreinauer) [reply examples/orbit_5k_pipeline.ipynb:196]: Also add a comment why we need a wrapped Persformer model

File: examples/orbit_5k_pipeline.ipynb [hunk 5]
```diff
@@ -155,14 +271,16 @@
...
     "batch = next(iter(dl_train))\n",
     "datum = batch[0][0].reshape(1, *(batch[0][0].shape))\n",
     "class_ = batch[1][0].item()\n",
     "\n",
-    "# we now the Saliency maps to interpret the results\n",
-    "inter = Interpreter(trainer.model, method=\"Saliency\")\n",
-    "\n",
     "# interpret the diagram\n",
     "x, attr = inter.interpret(x=datum, y=class_)\n",
     "\n",
```
Comment (raphaelreinauer) [inline examples/orbit_5k_pipeline.ipynb:286]: Add a new section explaining why you're doing hyperparameter search

File: examples/orbit_5k_pipeline.ipynb [hunk 6]
```diff
@@ -155,14 +271,16 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# get a datum and its corresponding class\n",
+    "# Initialize the Interpreter class in Saliency mode\n",
```
Comment (raphaelreinauer) [inline examples/orbit_5k_pipeline.ipynb:274]: Interpreter class -> Interpreter object

File: examples/orbit_5k_pipeline.ipynb [hunk 7]
```diff
@@ -130,23 +202,67 @@
...
+    "\n",
+    "One of the key feature of the Persformer achitecture, is that it is proven to satisfy a Universal Approximation Theorem. The authors of the [Persformer paper](https://arxiv.org/abs/2112.15210) also give evidence that this achitecture has less inductive bias than already existing deep neural networks architecture that can input persistence diagrams. This observation motivates the introduction of importance score of points in persistence diagrams in a classification task.\n",
+    " \n",
+    "The $\\textbf{Persformer}$ model for a classification problem is an almost everywhere differentiable function $F: \\mathcal D \\to \\mathbb{R}^m$, where $m$ is the number of classes and $\\mathcal D$ is the space of persistence diagrams. It maps a persistence diagram to the logits of the class probability. Let $d$ be the maximum homology dimension to be considered and let $x = (x_k)_{k\\in \\{ 1,\\ldots, n \\}}\\in (\\R^{2+d})^n$ be a persistence diagram and $i(x) = \\mathrm{argmax}_j F(x)_j$. The first two coordinates of $x_k \\in \\mathbb{R}^{2+d}$ are the birth and death coordinates and the last $d$ coordinates are the one-hot encoded homology dimensions. \n",
```
Comment (raphaelreinauer) [inline examples/orbit_5k_pipeline.ipynb:256]: \\R^{2+d})^n -> \mathbb

**Review by raphaelreinauer — CHANGES_REQUESTED** [review]
Some small changes. The most important one is preprocessing.
